pkg/web/apps/apps_service.go

```diff
@@ -34,3 +35,21 @@ func (a *appsService) GetApps() (*[]models.App, error) {
 
 	return apps, nil
 }
+
+// GetAppByID retrieves app by id from the repository
+func (a *appsService) GetAppByID(id string) (*models.App, error) {
+
+	app, err := a.repository.GetAppByID(id)
+
+	if err != nil {
+		return nil, err
+	}
+	if app == nil {
+		return nil, models.ErrNotFound
+	}
+
+	deployedVersions, _ := a.repository.GetAppVersionsByAppID(app.AppID)
```

Comment:
this could return an error as well that will need to be checked. There is also a scenario where no versions could exist for an app and that is valid and should not bubble an error to the HTTP layer.

Reply:
Still need it to return an app and not an error if no versions present, may be better to log an error here